### PR TITLE
L4.9.123 2.3.0 QCA

### PR DESCRIPTION
**Pull request by thochstein**

### CODE REVIEW

File: recipes-bsp/firmware-imx/firmware-imx_7.6.bb [hunk 1]
```diff
@@ -76,6 +76,27 @@ do_install() {
...
+    # Cadence HDMI
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmitxfw.bin ${DEPLOYDIR}
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmirxfw.bin ${DEPLOYDIR}
+    install -m 0644 ${S}/firmware/hdmi/cadence/dpfw.bin ${DEPLOYDIR}
+}
+do_deploy_append_mx8qxp() {
+    # SECO
+    install -m 0644 ${S}/firmware/seco/ahab-container.img ${DEPLOYDIR}
+}
+addtask deploy before do_build after do_install
```
Comment (otavio) [inline recipes-bsp/firmware-imx/firmware-imx_7.6.bb]: Please split those in specific recipes so we can build / install just the need and not clobber the deploy directory with unrelated files.

File: recipes-kernel/kernel-modules/kernel-module-qca6174_2.0.bb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require kernel-module-qcacld-lea.inc
+
+SUMMARY = "Qualcomm WiFi driver for QCA module 6174"
+
+EXTRA_OEMAKE += "${EXTRA_OEMAKE_QCA6174}"
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qca6174_2.0.bb]: Useless indirection, move it to here.

File: recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc [hunk 3]
```diff
@@ -43,6 +43,13 @@ EXTRA_OEMAKE_QCA9377 += " \
     MODNAME=qca9377 \
     SAP_AUTH_OFFLOAD=1 \
 "
+EXTRA_OEMAKE_QCA6174 = " \
+    CONFIG_ROME_IF=pci \
+    CONFIG_WLAN_FEATURE_11W=y \
+    CONFIG_WLAN_FEATURE_FILS=y \
+    CONFIG_WLAN_WAPI_MODE_11AC_DISABLE=y \
+    MODNAME=qca6174 \
+"
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc]: Move to .bb

File: recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    CONFIG_CLD_HL_SDIO_CORE=y \
+    CONFIG_FEATURE_COEX_PTA_CONFIG_ENABLE=y \
+    CONFIG_PER_VDEV_TX_DESC_POOL=1 \
+    CONFIG_QCA_LL_TX_FLOW_CT=1 \
+    CONFIG_QCA_SUPPORT_TXRX_DRIVER_TCP_DEL_ACK=y \
+    CONFIG_WLAN_FEATURE_FILS=y \
+    CONFIG_WLAN_WAPI_MODE_11AC_DISABLE=y \
+    MODNAME=qca9377 \
+    SAP_AUTH_OFFLOAD=1 \
+"
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc]: Move to .bb

File: recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    file://0010-CORE-HIF-PCIe-only-support-one-instance.patch \
+    file://0011-CORE-BMI-RF-align-the-utf-firmware-bin-name.patch \
+    file://0012-CORE-revert-two-patches-to-support-QCA-SMP.patch \
+    file://0013-CORE-services-os-fix-issue-to-avoid-spin_lock_bh-in-.patch \
+    file://0014-CLD-CORE-add-CONFIG_HDD_WLAN_WAIT_TIME-support-for-u.patch \
+    file://0015-CORE-HIF-enable-pcie-MSI-feature.patch \
+    file://0016-MLK-18490-CORE-HDD-add-ssid-len-check.patch \
+    file://0017-MLK-18491-qcacld-2.0-avoid-overflow-of-bounce-buffer.patch \
+    file://0018-MLK-18491-02-qcacld-2.0-fix-the-overflow-of-bounce-b.patch \
+"
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc]: NXP has the fork so please use it to avoid adding so many patches to the layer.

**Comment by thochstein** [comment]
I removed the first 3 commits to leave this pull request focused on QCA.

File: recipes-bsp/firmware-imx/firmware-imx_7.6.bb (hunk 1 above)
Comment (thochstein) [reply recipes-bsp/firmware-imx/firmware-imx_7.6.bb]: The deploy directory is machine-specific, so not sure what you mean here.

$ find tmp/deploy/ -name ahab-container.img
tmp/deploy/images/imx8qxpmek/ahab-container.img

File: recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc (hunk 5 above)
Comment (thochstein) [reply recipes-kernel/kernel-modules/kernel-module-qcacld-lea.inc]: I added the fork, but Lauren still has to push it to CodeAurora.

File: recipes-kernel/kernel-modules/kernel-module-qca9377_2.0.bb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    CONFIG_PER_VDEV_TX_DESC_POOL=1 \
+    CONFIG_QCA_LL_TX_FLOW_CT=1 \
+    CONFIG_QCA_SUPPORT_TXRX_DRIVER_TCP_DEL_ACK=y \
+    CONFIG_WLAN_FEATURE_FILS=y \
+    CONFIG_WLAN_WAPI_MODE_11AC_DISABLE=y \
+    MODNAME=qca9377 \
+    SAP_AUTH_OFFLOAD=1 \
+"
+
+RDEPENDS_${PN} += "firmware-qca9377"
```
Comment (thochstein) [inline recipes-kernel/kernel-modules/kernel-module-qca9377_2.0.bb]: The RDEPENDS for firmware-qca* and for qca-tools do not seem to have the intended effect. The resulting image has the kernel-module but is missing the firmware and the tools.

Comment (thochstein) [reply recipes-kernel/kernel-modules/kernel-module-qca9377_2.0.bb]: I added a commit to work around the problem with dependencies. Can you review the pull request and consider accepting it with the workaround?

**Comment by thochstein** [comment]
Ping, this is complete. The final commit is a workaround for the fact that RDEPENDS from kernel-module-qca* does not work as expected. Please take the commit so the workaround can be dealt with separately.
